### ISSUE TITLE
Powershell on remote target shows incomplete verbose logs

### ISSUE THREAD

**Issue by Ricconnect** [Microsoft/vsts-tasks#3406]
When I run remote powershell from a administrator powershell command prompt on the system of my buildserver, I get a more complete output than when I use the build in TFS task for Remote powershell in Release Management. 

When I use the TFS task, it seems to cut off the logging at some point, which is always somewhere on the same line but not specifically one line always. Is there something that cuts off the logging for the powershell remote? There isn't a difference between what is shown in the TFS portal and when I download the logs from the TFS portal. I use TFS 2015 Update 3.

**Comment by vincentdass** [comment]
@Ricconnect , Thanks for reporting the issue.
Can you share the last two release logs to **RM_Customer_Queries@microsoft.com** ?
Can u please queue and new release with `system.debug` = true and share those logs too ?

**Comment by Ricconnect** [comment]
@vincentdass  I have sent you the logs of the remote powershell task, together with a log of powershell when I manually do a Invoke Command remotely from the same pc as the release agent runs

**Comment by niadak** [comment]
@Ricconnect , IIS team has acknowledged that there is some issue with few commands due to which we are not getting the logs. For example, 'Get-WebSiteState' command has this issue and we can work around this by using (Get-WebSite  -Name "WebSiteName").State command instead.

Form logs it seems like we the first missing log is "Stopped application pool UMO-releaseManagement1". Can you confirm that the corresponding command is 'Stop-WebAppPool'. We have to inform IIS team about this command also.

**Comment by Ricconnect** [comment]
@niadak  I can confirm that I use the Stop-WebAppPool command

**Comment by niadak** [comment]
@Ricconnect Thanks for your confirmation. We will get back to you if we get any work around from IIS team.

**Comment by niadak** [comment]
@Ricconnect We have checked with IIS team. They acknowledged that Stop-WebAppPool command has the issue and suggested to use one of the following command as work around until they fix the issue.

Stop-WebItem IIS:\AppPools\{AppPoolName}
Or,
(get-item IIS:\AppPools\{AppPoolName}).Stop

Please try the above workaround to get unblocked. Please let us know if you are facing the issue for any other IIS command.

**Comment by Ricconnect** [comment]
@niadak Thank you for your reply. I indeed already refactored the script to use the WebAdministration module in other ways. The workaround for everyone is to use the Get-Item command on the IIS drive. With this I refactored the following commands, because they all stopped the output of the script:

- Get-WebAppPoolState
- Get-WebApplication
- Get-Website
- Stop-Website
- Start-WebAppPool

I hope this information is useful for you.
